### ISSUE TITLE
"TypeError: Descriptors cannot not be created directly." from Keras during installation

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#3801]
Our tests have started failing across all branches and workflows ([example](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/6608443416?check_suite_focus=true)) due to the following error:

```python
Check if Keras (2.3.1) is installed.................[FAIL]
An error occured while importing module Keras -> Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
Full traceback: Traceback (most recent call last):
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\scripts\sct_check_dependencies.py", line 271, in main
    module = module_import(module_name, suppress_stderr)
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\scripts\sct_check_dependencies.py", line 95, in module_import
    module = importlib.import_module(module_name)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\keras\__init__.py", line 3, in <module>
    from . import utils
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\keras\utils\__init__.py", line 6, in <module>
    from . import conv_utils
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\keras\utils\conv_utils.py", line 9, in <module>
    from .. import backend as K
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\keras\backend\__init__.py", line 1, in <module>
    from .load_backend import epsilon
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\keras\backend\load_backend.py", line 90, in <module>
    from .tensorflow_backend import *
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\keras\backend\tensorflow_backend.py", line 5, in <module>
    import tensorflow as tf
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow\__init__.py", line 102, in <module>
    from tensorflow_core import *
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\__init__.py", line 28, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "<frozen importlib._bootstrap>", line 1019, in _handle_fromlist
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow\__init__.py", line 50, in __getattr__
    module = self._load()
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow\__init__.py", line 44, in _load
    module = _importlib.import_module(self.__name__)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\python\__init__.py", line 52, in <module>
    from tensorflow.core.framework.graph_pb2 import *
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\core\framework\graph_pb2.py", line 16, in <module>
    from tensorflow.core.framework import node_def_pb2 as tensorflow_dot_core_dot_framework_dot_node__def__pb2
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\core\framework\node_def_pb2.py", line 16, in <module>
    from tensorflow.core.framework import attr_value_pb2 as tensorflow_dot_core_dot_framework_dot_attr__value__pb2
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\core\framework\attr_value_pb2.py", line 16, in <module>
    from tensorflow.core.framework import tensor_pb2 as tensorflow_dot_core_dot_framework_dot_tensor__pb2
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\core\framework\tensor_pb2.py", line 16, in <module>
    from tensorflow.core.framework import resource_handle_pb2 as tensorflow_dot_core_dot_framework_dot_resource__handle__pb2
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\core\framework\resource_handle_pb2.py", line 16, in <module>
    from tensorflow.core.framework import tensor_shape_pb2 as tensorflow_dot_core_dot_framework_dot_tensor__shape__pb2
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\tensorflow_core\core\framework\tensor_shape_pb2.py", line 42, in <module>
    serialized_options=None, file=DESCRIPTOR),
  File "d:\a\spinalcordtoolbox\spinalcordtoolbox\spinalcordtoolbox\venv_sct\lib\site-packages\google\protobuf\descriptor.py", line 560, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```

Note that https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738 will fix this issue by dropping Keras, so we may want to increase the priority of that one getting reviewed + approved. But, given that it's a larger PR, we may want to investigate a short-term fix that will allow us to continue development elsewhere.

**Comment by joshuacwnewton** [comment]
Upstream: https://github.com/protocolbuffers/protobuf/issues/10051

Proposed fix: https://github.com/protocolbuffers/protobuf/issues/10051#issuecomment-1138392640

> I've set protobuf==3.20.1 and it resolved the issue

We can temporarily set this now, and unpin it once #3738 is merged.

**Comment by joshuacwnewton** [comment]
It looks like the updated `protobuf` version was yanked, so we're no longer having issues on `master`. This can be closed now, I think.

**Comment by joshuacwnewton** [comment]
Huh. I just opened a new PR, and I'm seeing this again.

```
Downloading protobuf-4.21.0-cp37-abi3-manylinux2014_x86_64.whl 
```

I'm a bit confused as to why we started getting passing CI logs on `master`, but. I'm presuming that a fix still is needed after all.
